tiledb/sm/array_schema/array_schema.h

```diff
@@ -268,6 +268,11 @@ class ArraySchema {
   /** Return the pipeline used for coordinates. */
   const FilterPipeline& coords_filters() const;
 
+  /** Return the array current domain. */
+  inline const shared_ptr<CurrentDomain> current_domain() const {
```

Comment:
```suggestion
  inline const CurrentDomain& current_domain() const {
```
Unless we need to copy a `shared_ptr` for some reason, there's no need to pass it around as a function argument.